modules/virt-creating-a-service-from-a-running-virtual-machine.adoc

```diff
@@ -0,0 +1,191 @@
+
+[id="virt-creating-a-service-from-a-running-virtual-machine_{context}"]
+
+= Creating a service from a running virtual machine
+
+You can create and expose a service from a running virtual machine.
+
+Once the virtual machine is running, create a service object to expose the service.
+Connect to the service by specifying the service name.
+
+The service exposes the host name of the virtual machine to the internal cluster,
+using the `ClusterIP` service, or externally, using the `NodePort` or `LoadBalancer`
+service.
+
+Because the service is scoped by the namespace to which it is defined, ensure
+that you reference the correct namespace in all YAML definitions.
+
+This procedure presents an example of how to create and expose a service object
+of `type: ClusterIP` as a virtual machine-backed service.
+
+[NOTE]
+====
+ClusterIP is the default service `type`, if the service `type` is not specified.
+====
+
+.Procedure
+
+. Start the virtual machine if it is not already running.
+
+. Edit the `VirtualMachine` YAML configuration file to include the
+label `special: key`.
++
+
+[NOTE]
+====
+Labels on a `VirtualMachine` are passed through to the pod. The labels on
+the `VirtualMachine`, for example `special: key`, must match the labels on
+the Service object `selector` attribute.
+====
+
++
+
+[source,yaml]
+----
+apiVersion: kubevirt.io/v1alpha3
+kind: VirtualMachine
+metadata:
+  name: vm-ephemeral
+  labels:
+    special: key <1>
+    spec:
+      domain:
+        devices:
+          disks:
+            - name: containerdisk
+              disk:
+                bus: virtio
+            - name: cloudinitdisk
+              disk:
+                bus: virtio
+          interfaces:
+          - masquerade: {}
+            name: default
+        resources:
+          requests:
+            memory: 1024M
+      networks:
+        - name: default
+          pod: {}
+      volumes:
+        - name: containerdisk
+          containerDisk:
+            image: kubevirt/fedora-cloud-container-disk-demo
+        - name: cloudinitdisk
+          cloudInitNoCloud:
+            userData: |
+              #!/bin/bash
+              echo "fedora" | passwd fedora --stdin
+----
+<1> The `special: key` label
++
+
+. Edit the `VirtualMachine` YAML to create and expose the `Service` object, connecting by
+specifying the service name.
+
+.. Add `type: ClusterIP` for a
+ClusterIP service. Add `targetPort: 22`, exposing the service on SSH port `22`.
++
+To create and expose other types of services, such as `NodePort` and `LoadBalancer`, replace
+`type: ClusterIP` with `type: NodePort` or `type: LoadBalancer`, as appropriate,
+in the Service object definition.
++
+.. Specify a `name` for the service you are creating and exposing.
++
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: vmservice <1>
+spec:
+  ports:
+  - port: 27017
+    protocol: TCP
+    targetPort: 22 <2>
+  selector:
+    special: key
```

Comment:
I think it's worth adding legend bullet <4> (sorry, I don't know what is the official term in documentation terminology) and say that this selector corresponds to the label added in the VM.

Reply:
Great, thanks Yossi!